File: Completed/JavaScript/362.js
/**
 * Initialize your data structure here.
 */
var HitCounter = function() {
  this.map = {}
};

/**
* Record a hit.
      @param timestamp - The current timestamp (in seconds granularity). 
* @param {number} timestamp
* @return {void}
*/
HitCounter.prototype.hit = function(timestamp) {
  if (this.map[timestamp]) {
      this.map[timestamp] += 1
  } else {
      this.map[timestamp] = 1
  }
};

/**
* Return the number of hits in the past 5 minutes.
      @param timestamp - The current timestamp (in seconds granularity). 
* @param {number} timestamp
* @return {number}
*/
HitCounter.prototype.getHits = function(timestamp) {
  let result = 0
  
  for (let key of Object.keys(this.map)) {
      const time = Number(key)
      if (timestamp - 300 < time && time <= timestamp) {
          result += this.map[key]
      }
  }
  
  return result
};

/** 
* Your HitCounter object will be instantiated and called as such:
* var obj = new HitCounter()
* obj.hit(timestamp)
* var param_2 = obj.getHits(timestamp)
*/

File: Completed/Go/362.go
type HitCounter struct {
    counts map[int]int
}


func Constructor() HitCounter {
    return HitCounter{
        map[int]int{},
    }
}


func (this *HitCounter) Hit(timestamp int)  {
    this.counts[timestamp]++
}


func (this *HitCounter) GetHits(timestamp int) int {
    res := 0
    for i := timestamp; i > timestamp-300; i-- {
        if i < 0 {
            break
        }
        
        res += this.counts[i]
    }
    return res
}


/**
 * Your HitCounter object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Hit(timestamp);
 * param_2 := obj.GetHits(timestamp);
 */
